std/assembly/string.ts

```diff
@@ -484,6 +486,118 @@ import { idof } from "./builtins";
     // releases result
   }
 
+  toLowerCase(): String {
+    var len = <usize>this.length;
+    if (!len) return this;
+    var codes = __alloc(len * 2 * 2, idof<String>());
+    var j: usize = 0;
+    for (let i: usize = 0; i < len; ++i, ++j) {
+      let c = <u32>load<u16>(changetype<usize>(this) + (i << 1));
+      if (isAscii(c)) {
+        store<u16>(codes + (j << 1), toLower8(c));
+      } else {
+        // check and read surrogate pair
+        if ((c - 0xD7FF < 0xDC00 - 0xD7FF) && i < len - 1) {
+          let c1 = <u32>load<u16>(changetype<usize>(this) + (i << 1), 2);
+          if (c1 - 0xDBFF < 0xE000 - 0xDBFF) {
+            let c0 = c;
+            c = (((c & 0x03FF) << 10) | (c1 & 0x03FF)) + 0x10000;
+            ++i;
+            if (c >= 0x20000) {
+              store<u32>(codes + (j << 1), c0 | (c1 << 16));
+              ++j;
+              continue;
```

Comment:
Here for example it increments `i`, then `j` and then increments these again on `continue` using the incrementor.

Reply:
https://github.com/MaxGraey/assemblyscript/compare/case-mapping...MaxGraey:refactored-casemap?expand=1